### ISSUE TITLE
Filtering by Related Collections not working

### ISSUE THREAD

**Issue by LennardF1989** [directus/directus#24007]
### Describe the Bug

I have a database with quite a few tables all with proper FK's to eachother. I've also added Data Models for these tables and configured the related fields accordingly.

However, while the Data Model recognizes there is a relationship, eg:
<img width="457" alt="image" src="https://github.com/user-attachments/assets/80abba79-9b4b-494a-a3ac-946fe058707c">

When I try to filter on a property of the Domains table, I do not get the expansion icon in the Filter UI:
<img width="385" alt="image" src="https://github.com/user-attachments/assets/ec066119-9138-49be-bb8a-4d0f0edad634">

This is the case for all tables and their relationships.

I cannot find any documentation on whether or not this is something I have to physically enable somewhere, so this might be something that worked before, but doesn't anymore, or a bug with Microsoft SQL, perhaps?

Related documentation: https://docs.directus.io/user-guide/content-module/filters.html#filtering-by-related-collections

### To Reproduce

1. Have the following table structure in Microsoft SQL (but probably any will do):
```
SET ANSI_NULLS ON
GO
SET QUOTED_IDENTIFIER ON
GO
CREATE TABLE [dbo].[Domains](
	[Id] [int] IDENTITY(1,1) NOT NULL,
	[Alias] [varchar](20) NOT NULL,
	[LastSyncTimestamp] [int] NOT NULL,
	[SyncEnabled] [bit] NOT NULL,
	[ProjectDefinition] [varchar](255) NOT NULL,
	[Archived] [bit] NOT NULL,
 CONSTRAINT [PK_Domains] PRIMARY KEY CLUSTERED 
(
	[Id] ASC
)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON, OPTIMIZE_FOR_SEQUENTIAL_KEY = OFF) ON [PRIMARY]
) ON [PRIMARY]
GO
SET ANSI_NULLS ON
GO
SET QUOTED_IDENTIFIER ON
GO
CREATE TABLE [dbo].[Groups](
	[Id] [int] IDENTITY(1,1) NOT NULL,
	[ParentId] [int] NULL,
	[Name] [varchar](255) NOT NULL,
	[DomainId] [int] NULL,
	[Archived] [bit] NOT NULL,
 CONSTRAINT [PK_Groups] PRIMARY KEY CLUSTERED 
(
	[Id] ASC
)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON, OPTIMIZE_FOR_SEQUENTIAL_KEY = OFF) ON [PRIMARY]
) ON [PRIMARY]
GO
ALTER TABLE [dbo].[Groups]  WITH CHECK ADD  CONSTRAINT [FK_Groups_Domains_DomainId] FOREIGN KEY([DomainId])
REFERENCES [dbo].[Domains] ([Id])
GO
ALTER TABLE [dbo].[Groups] CHECK CONSTRAINT [FK_Groups_Domains_DomainId]
GO
ALTER TABLE [dbo].[Groups]  WITH CHECK ADD  CONSTRAINT [FK_Groups_Groups_ParentId] FOREIGN KEY([ParentId])
REFERENCES [dbo].[Groups] ([Id])
GO
ALTER TABLE [dbo].[Groups] CHECK CONSTRAINT [FK_Groups_Groups_ParentId]
GO
```
2. Create a new instance of Directus with a Microsoft SQL connection.
3. Create the Data Models
4. Try filtering on the Domain ID under Groups.

### Directus Version

11.1.1

### Hosting Strategy

Self-Hosted (Custom)

### Database

Microsoft SQL

**Comment by LennardF1989** [comment]
I get the following error, which is odd since DomainId.Archived should be a working path.
```{"errors":[{"message":"You don't have permission to access field \"DomainId.Archived\" in collection \"Groups\" or it does not exist. Queried in root.","extensions":{"reason":"You don't have permission to access field \"DomainId.Archived\" in collection \"Groups\" or it does not exist. Queried in root.","code":"FORBIDDEN"}}]}```

**Comment by LennardF1989** [comment]
No, it's a full admin user. I'm the only user even.

I've played some more and the following query gave me data:
`GET /items/Groups?fields[]=DomainId.Archived`

When I do this, it also gives me data:
`GET /items/Groups?fields[]=DomainId.Archived&filter[DomainId][_eq]=1`

However, when I try to use the same field in a filter like you proposed, I get the error I mentioned earlier:
`GET /items/Groups?fields[]=DomainId.Archived&filter[DomainId.Archived][_eq]=true`

**Comment by hanneskuettner** [comment]
I think @br41nslug mixed something up here and the filter you want o use is 

`GET /items/Groups?fields[]=DomainId.Archived&filter[DomainId][Archived][_eq]=true`

**Comment by LennardF1989** [comment]
I just debugged a little and found this function that builds the filter UI: https://github.com/directus/directus/blob/d0c74917af60cb894e3bc1cdec6c3a9e73722879/app/src/components/v-field-list/v-field-list.vue#L62

`relationCollection` always seems to be `undefined`:
<img width="284" alt="image" src="https://github.com/user-attachments/assets/48526927-e30e-4008-8ed6-6b3f16cd4e84">

**Comment by hanneskuettner** [comment]
How did you set up the relation within Directus? It should be a M2O collection to the `Domains` collection, I assume?

**Comment by LennardF1989** [comment]
Yes, it's set as a M2O. It mentions that in the UI before you click the field. It's also visible on the screenshot in the first post.

Still digging, somewhere somehow information is lost.

**Comment by LennardF1989** [comment]
Okay, getting somewhere, the cullprit appears to be this line: https://github.com/directus/directus/blob/d0c74917af60cb894e3bc1cdec6c3a9e73722879/app/src/composables/use-field-tree.ts#L144

Somehow, it recognizes the M2O relationship, but the `meta` is always `undefined` - so it cancels out early.

**Comment by LennardF1989** [comment]
Okey, yea. Something is wrong on the backend side of things, not the UI.

Whenever the store gets its data from /relations, there is no `meta` for this particular relation for some reason:
```
   {
      "collection": "Groups",
      "field": "DomainId",
      "related_collection": "Domains",
      "schema": {
        "table": "Groups",
        "column": "DomainId",
        "foreign_key_schema": "dbo",
        "foreign_key_table": "Domains",
        "foreign_key_column": "Id",
        "constraint_name": "FK_Groups_Domains_DomainId",
        "on_update": "NO ACTION",
        "on_delete": "NO ACTION"
      },
      "meta": null
    },
```

Circling back, this may be a MSSQL issue.

**Comment by LennardF1989** [comment]
Okey, I think I've found the issue. I've exported a snapshot from my local PC to this machine I'm testing on. The .yaml 100% has the relations in the `directus_relations` table. However, the machine where I imported this snapshot, does not.

What I did was:
1. Create a clean Directus instance and setup the admin account
2. Apply the snapshot

So this bug is related to the `schema apply`.

**Comment by LennardF1989** [comment]
Yup, found it! Most likely this is broken for everyone, not just MSSQL.

**Issue #1**
Here's what happens: https://github.com/directus/directus/blob/main/api/src/cli/commands/schema/apply.ts#L63C33-L63C44

`getSnapshot` will always run `RelationService.readAll`. Even if `directus_relations` is empty, it will still succeed in getting some data from a clean database due to `schemaInspector` - albit without any `meta`.

Next when `getSnapshotDiff` will try to do its thing. It will realize that the data in the .yaml is already the same as whatever `RelationService.readAll` reports. So it will never actually get into the logic of the `applyDiff` function to loop over the relations and trigger `RelationService.createOne`.

**Issue #2**
This brings up another issue: why is `directus_relations` always empty in my situation?

This problem comes from pre-existing tables, again related to the `RelationService.readAll`. If you convert a from a database table to one maintained by Directus, it will again ask the RelationService for information. Since the SchemaInspector is more than happy to respond, this line here will always be `true`: https://github.com/directus/directus/blob/main/app/src/stores/relations.ts#L32

As such, no matter what you do: if you had a pre-existing table in your database, you will never be able to actually create the `directus_relations` entry!

**Fix?**
This is a hard one... In these situations, the result of the SchemaInspector should be ignored and the table/fields treated as a new table to be created...

*Workaround*
The workaround is not something easily done right now, since there is no API exposing JUST the `relationsItemService.createOne` function.

The data to insert for each relation is:
```
const metaRow = {
  ...(relation.meta || {}),
  many_collection: relation.collection,
  many_field: relation.field,
  one_collection: relation.related_collection || null,
};
```
